crates/ruff/src/rules/flake8_bugbear/rules/no_explicit_stacklevel.rs

```diff
@@ -0,0 +1,68 @@
+use rustpython_parser::ast::{Expr, Keyword};
+
+use ruff_diagnostics::{Diagnostic, Violation};
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_ast::helpers::SimpleCallArgs;
+use ruff_python_ast::types::Range;
+
+use crate::checkers::ast::Checker;
+
+/// ## What it does
+/// Checks for `warnings.warn` calls without an explicit `stacklevel` keyword
+/// argument.
+///
+/// ## Why is this bad?
+/// The `warnings.warn` method uses a `stacklevel` of 1 by default, which
+/// limits the rendered stack trace to that of the line on which the
+/// `warn` method is called.
+///
+/// It's recommended to use a `stacklevel` of 2 or higher, give the caller
+/// more context about the warning.
+///
+/// ## Example
+/// ```python
+/// warnings.warn("This is a warning")
+/// ```
+///
+/// Use instead:
+/// ```python
+/// warnings.warn("This is a warning", stacklevel=2)
+/// ```
+#[violation]
+pub struct NoExplicitStacklevel;
+
+impl Violation for NoExplicitStacklevel {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        format!("No explicit `stacklevel` keyword argument found")
```

Comment:
I truncated this message a bit and instead moved the additional context into docs.